modules/caddyhttp/replacer.go

```diff
@@ -470,7 +469,11 @@ func marshalPublicKey(pubKey any) ([]byte, error) {
 	case *rsa.PublicKey:
 		return asn1.Marshal(key)
 	case *ecdsa.PublicKey:
-		return elliptic.Marshal(key.Curve, key.X, key.Y), nil
+		e, err := key.ECDH()
+		if err != nil {
+			return nil, err
+		}
+		return e.Bytes(), nil
```

Comment:
`elliptic.Marshal` is deprecated. I'm not sure if this change is correct, wouldn't mind a second pair of eyes. But I think this is the intended use of the new API?